pkg/azure/utils.go

```diff
@@ -648,11 +648,11 @@ func waitForDataDiskDetachment(ctx context.Context, clients spi.AzureDriverClien
 
 		future, err := clients.GetVM().CreateOrUpdate(ctx, resourceGroupName, *vm.Name, vm)
 		if err != nil {
-			return OnARMAPIErrorFail(prometheusServiceVM, err, "Failed to CreateOrUpdate. Error Message - %s", err)
+			return OnARMAPIErrorFail(prometheusServiceVM, fmt.Errorf("Failed to detach data disc: %w", err), "Failed to CreateOrUpdate. Error Message - %s", err)
```

Comment:
I think it is safer to do such as:

```suggestion
			return OnARMAPIErrorFail(prometheusServiceVM, err, "Failed to detach data disk from VM. Error Message - %s", err)
```

As far I see the err is passed to `OnARMAPIErrorFail` then to `OnErrorFail` and then some requests IDs are extracted from the error. If we wrap it like this, I am not sure whether we won't break this handling.

Reply:
Nice catch on the breaking of the error handling when we wrap the error. `OnARMAPIErrorFail ` does not modify the second parameter error which is returned by the function. Third and fourth parameters are used only for logging. Therefore the second parameter is the error that gets displayed on the dashboard.
What do you think about this change:
``` golang
func RetrieveRequestID(err error) (bool, string, *autorest.DetailedError) {
	var detailedErr autorest.DetailedError
	if errors.As(err, &detailedErr) && detailedErr.Response != nil {
		requestID := strings.Join(detailedErr.Response.Header["X-Ms-Request-Id"], "")
		return true, requestID, &detailedErr
	}
	return false, "", nil
}
```
`errors.As` is invariant to how many times the expected error type has been wrapped. Therefore with the function change above and my initial code we will handle request ID extraction + dashboard error message would be more informative